File: sort/1528_shuffle_string/1528_shuffle_string.go
package main

import "fmt"

func restoreString(s string, indices []int) string {
	var b [100]byte
	for i, v := range indices {
		b[v] = s[i]
	}
	return string(b[:len(s)])
}

func main() {
	s := "codeleet"
	indices := []int{4, 5, 6, 7, 0, 2, 1, 3}
	res := restoreString(s, indices)
	fmt.Println(res)
}

File: sort/1528_shuffle_string/1528_shuffle_string.py
from typing import List

class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        _s = ['0'] * len(s)
        for idx, val in enumerate(s):
            _s[indices[idx]] = val
        return ''.join(_s)


s = "codeleet"
indices = [4,5,6,7,0,2,1,3]
res = Solution().restoreString(s, indices)
print(res)